src/style/style.js

```diff
@@ -521,6 +521,11 @@ class Style extends Evented {
 
         const id = layerObject.id;
 
+        if (this.getLayer(id)) {
+            this.fire('error', {error: new Error(`Layer with id "${id}" already exists on this map`)});
```

Comment:
`Map.addSource`  will throw an error when attempting to add a source with a duplicate ID. Is there a reason to fire an error event instead of throwing for `addLayer` ?

Reply:
Firing an event is correct; see #3879.

Reply:
We made a decision awhile back to emit errors across the board instead of throwing https://github.com/mapbox/mapbox-gl-js/issues/3879#issue-198365050

Reply:
whoops didn't reload and see john's comment 😬